doc/object_lifetime_management.md

```diff
@@ -0,0 +1,80 @@
+# Object lifetime management
+
+As the methods and classes within the node-addon-api are used,
+handles to objects in the heap for the underlying
+VM may be created. A handle may be created for any new node-addon-api
+Value(and subclasses) created or returned. These handles must hold the
```

Comment:
A handle may be created when any new node-addon-api Value and its subclasses have been created or returned.